partition/utils.go

```diff
@@ -26,38 +26,22 @@ import (
 	"strings"
 )
 
-// FIXME: would it make sense to differenciate between launch errors and
-//        exit code? (i.e. something like (returnCode, error) ?)
-func runCommandImpl(args ...string) (err error) {
-	if len(args) == 0 {
-		return errors.New("no command specified")
-	}
-
-	if out, err := exec.Command(args[0], args[1:]...).CombinedOutput(); err != nil {
-		cmdline := strings.Join(args, " ")
-		return fmt.Errorf("Failed to run command '%s': %s (%s)",
-			cmdline, out, err)
-	}
-	return nil
-}
-
-// Run the command specified by args
 // This is a var instead of a function to making mocking in the tests easier
-var runCommand = runCommandImpl
+var runCommandWithStdout = runCommandWithStdoutImpl
 
 // Run command specified by args and return the output
-func runCommandWithStdoutImpl(args ...string) (output string, err error) {
+func runCommandWithStdoutImpl(args ...string) (string, error) {
 	if len(args) == 0 {
+
 		return "", errors.New("no command specified")
 	}
 
-	bytes, err := exec.Command(args[0], args[1:]...).Output()
+	output, err := exec.Command(args[0], args[1:]...).CombinedOutput()
```

Comment:
It's usually a bad idea to grab the stderr when the output is oriented towards machine-parsing, unless you're specifically looking to pick errors up from that as well. If what you're looking for is error-reporting with proper error from the command (a good idea!), I'd suggest instead picking up both by setting `Stdout` and `Stderr` to `bytes.Buffer` values, returning stdout alone on success, and stderr inside the error value on failures. I'd also keep the original function name for both (`runCommand{,Impl}`).